mint-client/src/bin/clientd/main.rs

```diff
@@ -33,43 +25,153 @@ struct Opts {
     workdir: PathBuf,
 }
 
+/*
+#[derive(Debug, Deserialize, Serialize)]
+struct InfoResponse {
+    total : serde_json::Value,
+    coins : serde_json::Value,
+}
+*/
+#[derive(Serialize)]
+struct InfoResponse {
+    total : CoinTotal,
+    coins : Vec<CoinGrouped>,
+}
+#[derive(Serialize)]
+struct CoinTotal{
+    coin_count : usize,
+    amount : Amount,
+}
+#[derive(Serialize)]
+struct CoinGrouped{
+    amount : usize,
+    tier : u64
+}
+#[derive(Serialize)]
+struct SpendResponse {
+    token : String,
+}
+
+#[derive(Serialize)]
+struct ReissueResponse {
+    id : String,
+    fetched : Vec<TransactionId>,
+}
+
+#[derive(Deserialize,Serialize, Debug)]
+struct ReqBody<T> {
+    //I dont like this
+    value : T
+}
+
+impl InfoResponse {
+    fn new(coins: Coins<SpendableCoin>) -> Self {
+        let info_total =  CoinTotal { coin_count: coins.coin_count(), amount: coins.amount() };
+        let info_coins : Vec<CoinGrouped> = coins.coins.iter()
+            .map(|(tier, c)| CoinGrouped { amount : c.len(), tier : tier.milli_sat})
+            .collect();
+        InfoResponse { total : info_total, coins : info_coins }
+    }
+}
+
+impl ReissueResponse {
+    async fn new(mint_client :&MintClient, coins:Coins<SpendableCoin>) -> Self {
+        let mut rng = rand::rngs::OsRng::new().unwrap();
+        let id = mint_client.reissue(coins, &mut rng).await.unwrap();
+
+        let  id = id.to_hex();
+        let  fetched : Vec<TransactionId>= mint_client.fetch_all_coins().await.unwrap();
+        let res = ReissueResponse {
+            id,
+            fetched,
+        };
+        res
+    }
+}
 #[tokio::main]
 async fn main() -> tide::Result<()>{
-    //Instanciate CLient
+    //collect trace to print info
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
+        )
+        .init();
+
+
+    //Instantiate Client
     let opts: Opts = StructOpt::from_args();
     let cfg_path = opts.workdir.join("client.json");
     let db_path = opts.workdir.join("client.db");
-    //why use Config in ln-gateway
     let cfg: ClientConfig = load_from_file(&cfg_path);
     let db = sled::open(&db_path)
-        .unwrap()
+        .unwrap() //handle error ?
```

Comment:
Yes, handling errors here would be necessary for a final version. I left it like that because for me the message was informative enough whenever it failed (forgot to create directory for example). I don't have much experience with application-level error handling, maybe log as `error!("message …")` and then `std::process::exit` (or just return if in `main`)?

Generally, any `unwrap` means "I didn't think about it enough, needs to be revisited" whereas `expect` should describe why something is ok to ignore (except for the `DB error` thing, which means "if we can't even read from disk anymore let's just fail").